commons/src/main/java/io/aiven/kafka/tieredstorage/commons/ChunkManager.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2023 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.tieredstorage.commons;
+
+import java.io.InputStream;
+import java.util.concurrent.Future;
+
+import org.apache.kafka.server.log.remote.storage.RemoteLogSegmentMetadata;
+
+import io.aiven.kafka.tieredstorage.commons.manifest.SegmentManifest;
+
+public interface ChunkManager {
+    /**
+     * Gets a chunk of a segment.
+     * @return an {@link InputStream} of the chunk, plain text (i.e. decrypted and decompressed).
+     */
+    Future<InputStream> getChunk(RemoteLogSegmentMetadata remoteLogSegmentMetadata,
```

Comment:
Could you also elaborate (or add in the doc) why is this async?

AFAIK, the TS framework already wraps remote fetchs on Future:

https://github.com/apache/kafka/blob/a0a17208ad5ac57d35c5bd2c7ec2836ed3cafe66/core/src/main/java/kafka/log/remote/RemoteLogManager.java#L813-L815

Reply:
That's a good question. The `Future` approach comes from some early prototyping we did with @AnatolyPopov where it's possible to satisfy multiple identical parallel `getChunk` requests without doing downloading the same segment multiple times. `Future` worked as a very convenient placeholder and synchronization barrier. This could have been done on locks and atomic references, but futures are just more comfortable.

I propose to treat this `ChunkManager` interface as a starting point and change it as we see fit when the real implementation is done, including also removing the `Future` wrapper.